### PR TITLE
fix dependabot dependencies and suppression

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

fix dependabot dependencies and suppression

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -243,18 +243,18 @@ dependencies {
     exclude group: 'junit', module: 'junit'
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
   }
-  testImplementation group: 'com.azure', name: 'azure-core-test', version: '1.1.0', {
+  testImplementation group: 'com.azure', name: 'azure-core-test', version: '1.2.1', {
```
Comment (doncem) [inline build.gradle:246]: i don't think we are using it at all ;)

File: build.gradle [hunk 2]
```diff
@@ -243,18 +243,18 @@ dependencies {
...
     exclude group: 'com.github.tomakehurst', module: 'wiremock-standalone'
   }
 
   testCompile group: 'org.apache.commons', name: 'commons-email', version: '1.5'
 
   integrationTestImplementation sourceSets.main.runtimeClasspath
   integrationTestImplementation sourceSets.test.runtimeClasspath
-  integrationTestImplementation group: 'com.azure', name: 'azure-core-test', version: '1.1.0', {
+  integrationTestImplementation group: 'com.azure', name: 'azure-core-test', version: '1.2.1', {
```
Comment (doncem) [inline build.gradle:254]: weird. shouldn't be here if it is in `testImplementation` scope

File: build.gradle (hunk 1 above)
Comment (mdayican) [reply build.gradle:246]: we are using. but as I see we can get rid of. we used in mock.

Comment (doncem) [reply build.gradle:246]: so yea - we are not using it :D

File: build.gradle (hunk 2 above)
Comment (mdayican) [reply build.gradle]: I thought it is on purpose. removed it.

File: build.gradle (hunk 1 above)
Comment (mdayican) [reply build.gradle:246]: we are using a mock object from the lib.

**Pull request merged**